viewer/packages/tools/src/Timeline/Keyframe.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import { NodeCollectionBase, NodeAppearance } from '@reveal/core/src';
+import { Cognite3DModel } from '@reveal/core';
+
+/**
+ * Timeline Key Frames contains parameters to access Nodes, Styles for the Timeline
+ */
+export class Keyframe {
+  private readonly _date: Date;
+  private readonly _model: Cognite3DModel;
+  private readonly _nodeCollectionAndAppearnace: { nodes: NodeCollectionBase; nodeAppearance: NodeAppearance }[] = [];
+
+  constructor(model: Cognite3DModel, date: Date) {
+    this._model = model;
+    this._date = date;
+  }
+
+  /**
+   * Get date of the Keyframe
+   * @returns date
+   */
+  public getKeyframeDate(): Date {
+    return this._date;
+  }
+
+  /**
+   * Assigns the styles for the node set for the model for this Keyframe
+   */
+  public activate() {
+    this._nodeCollectionAndAppearnace.forEach((node, _index) => {
+      this._model.assignStyledNodeCollection(node.nodes, node.nodeAppearance);
+    });
+  }
+
+  /**
+   * Removes the style for the model
+   */
+  public deactivate() {
+    this._nodeCollectionAndAppearnace.forEach((node, _index) => {
+      this._model.unassignStyledNodeCollection(node.nodes);
+    });
+  }
+
+  /**
+   * Add node & style to the collection
+   * @param nodeCollection Node set to apply the Styles
+   * @param nodeAppearance Style to assign to the node collection
+   */
+  public assignStyledNodeCollection(nodeCollection: NodeCollectionBase, nodeAppearance: NodeAppearance) {
+    this._nodeCollectionAndAppearnace.push({ nodes: nodeCollection, nodeAppearance });
+  }
+
+  /**
+   * Remove Node & Style for this keyframe's nodeCollection and nodeAppearance
+   * @param nodeCollection Nodes to be unassign from node collection
+   */
+  public unassignStyledNodeCollection(nodeCollection: NodeCollectionBase) {
+    let index = this._nodeCollectionAndAppearnace.findIndex(x => x.nodes === nodeCollection);
+    while (index !== -1) {
+      this._nodeCollectionAndAppearnace.splice(index, 1);
+      index = this._nodeCollectionAndAppearnace.findIndex(x => x.nodes === nodeCollection);
+    }
```

Comment:
This behaviour has been changed a bit in `Cognite3dModel` last week - can you have a look at the changes and make the behaviour reflect that? Basically it's just that a nodeCollection only can have one style.